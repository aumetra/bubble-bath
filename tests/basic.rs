use bubble_bath::BubbleBath;
use insta::assert_snapshot;

#[test]
fn script() {
    assert_snapshot!(bubble_bath::clean(
        r#"
            <script>alert('XSS!');</script>
            <p>Hello world!</p>
            <font size="20">LARGE</font>
        "#
    )
    .unwrap());
}

#[test]
fn preserve_escaped() {
    let bubble_bath = BubbleBath {
        preserve_escaped: true,
        ..BubbleBath::default()
    };

    assert_snapshot!(bubble_bath
        .clean(
            r#"
            <p>Hello world!</p>
            <font size="20">LARGE</font>
        "#
        )
        .unwrap());
}
