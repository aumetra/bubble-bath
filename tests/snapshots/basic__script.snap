---
source: tests/basic.rs
expression: "bubble_bath::clean(r#\"\n            <script>alert('XSS!');</script>\n            <p>Hello world!</p>\n            <font size=\"20\">LARGE</font>\n        \"#).unwrap()"
---

            
            <p>Hello world!</p>
            LARGE
        
